src/common/function/src/scalars/timestamp/greatest.rs

```diff
@@ -87,6 +117,47 @@ impl Function for GreatestFunction {
                     zip::zip(&boolean_array, &column1, &column2).context(ArrowComputeSnafu)?;
                 Ok(Helper::try_into_vector(&result).context(error::FromArrowArraySnafu)?)
             }
+            ConcreteDataType::DateTime(_) => {
+                let column1 = columns[0].to_arrow_array();
+                let column1 = column1.as_primitive::<Date64Type>();
+                let column2 = columns[1].to_arrow_array();
+                let column2 = column2.as_primitive::<Date64Type>();
+                let boolean_array = gt(&column1, &column2).context(ArrowComputeSnafu)?;
+                let result =
+                    zip::zip(&boolean_array, &column1, &column2).context(ArrowComputeSnafu)?;
+                Ok(Helper::try_into_vector(&result).context(error::FromArrowArraySnafu)?)
+            }
+            ConcreteDataType::Timestamp(ts_type) => {
+                let column1 = columns[0].to_arrow_array();
+                let column2 = columns[1].to_arrow_array();
+
+                let boolean_array = match ts_type {
+                    TimestampType::Second(_) => {
```

Comment:
Can we use macro to simplify thos match arms?

Reply:
refactor at https://github.com/GreptimeTeam/greptimedb/pull/5005/commits/657e992a5a73282e31f4a944170f8aa575bf908c